app/models/case_log.rb

```diff
@@ -327,6 +356,9 @@ def set_derived_fields!
       self.wtcharge = weekly_value(tcharge) if tcharge.present?
     end
     self.has_benefits = get_has_benefits
+    self.wtshortfall = if tshortfall && receives_housing_related_benefits?
+                         weekly_value(tshortfall)
+                       end
```

Comment:
Do we want to reset it to nil in the `else` case?

Reply:
It will do that already since it'll assign whatever is returned which will implicitly be nil (there is a spec proving that).